app/main.py

```diff
@@ -2,24 +2,79 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: int, value: int) -> str:
+        self.value = value
+
+    def __get__(self) -> bool:
+        try:
+            if self.limitation_class == ChildrenSlideLimitationValidator:
+                if (
+                        not (4 <= self.age <= 14)
+                        or not (80 <= self.height <= 120)
+                        or not (20 <= self.weight <= 50)
+                ):
+                    raise ValueError(
+                        "Invalid parameters for accessing children slide"
+                    )
+            elif self.limitation_class == AdultSlideLimitationValidator:
+                if (
+                        not (14 <= self.age <= 60)
+                        or not (120 <= self.height <= 220)
+                        or not (50 <= self.weight <= 120)
+                ):
+                    raise ValueError(
+                        "Invalid parameters for accessing adult slide"
+                    )
+            return True
+        except ValueError:
+            return False
```

Comment:
If I will have 100 Validators? Each will works by if-else?
Think how you can use min_amount and max_amount

Reply:
Also it must be into `__set__` get must return current value. Please rewatch video about descriptors